lib/packwerk/node_processor_builder.rb

```diff
@@ -0,0 +1,92 @@
+# typed: true
+# frozen_string_literal: true
+
+require "active_support/inflector"
+require "constant_resolver"
+
+require "packwerk/association_inspector"
+require "packwerk/constant_discovery"
+require "packwerk/const_node_inspector"
+require "packwerk/dependency_checker"
+require "packwerk/file_processor"
+require "packwerk/node_processor"
+require "packwerk/package_set"
+require "packwerk/privacy_checker"
+require "packwerk/reference_extractor"
+
+module Packwerk
+  class NodeProcessorBuilder < T::Struct
+    extend T::Sig
+
+    DEFAULT_CHECKERS = [
+      ::Packwerk::DependencyChecker,
+      ::Packwerk::PrivacyChecker,
+    ]
+
+    const :root_path, String
+    const :reference_lister, T.nilable(ReferenceLister)
+    const :package_paths, T.nilable(String)
+    const :node_processor_class, T.untyped, default: ::Packwerk::NodeProcessor
+    const :load_paths, T.nilable(T::Array[String])
+    const :inflector, T.nilable(T.untyped)
+    const :custom_associations, T::Array[String], default: []
+    const :checker_classes, T.untyped, default: DEFAULT_CHECKERS
```

Comment:
@paracycle I couldn't find it in the docs, but do you know if we can specify a class and not an object as a type in sorbet?

Reply:
I think you want [`T.class_of`](https://sorbet.org/docs/class-of)